articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,563 @@
+---
+title: Run Q# standalone or with host programs
+description: Overview of the different ways to run Q# programs: command line, Q# Jupyter Notebooks, and classical host programs in Python or a .NET language.
+author: gillenhaalb
+ms.author: a-gibec@microsoft.com
+ms.date: 05/15/2020
+ms.topic: article
+uid: microsoft.quantum.guide.host-programs
+---
+
+# Run Q# standalone or with host programs
```

Comment:
I don't know if having Standalone in the title is the best option, if we have Command Line Applications in the installation guide. We should try to go with one to avoid user's confusion.

Reply:
@geduardo @KittyYeungQ @cgranade Only thing left for this PR is needing consensus on the title. If we want it to be something along these lines, I don't think we should specifically refer to the comparison as between the command line and host programs, because that completely excludes Q# Jupyter NBs. Hence the use of "standalone" in an attempt to include it (though not sure if that's valid). 

Alternatively, we could go in the less-specific direction and consider something like "Ways to call a Q# program", but I think that would open the door to more confusion.

Thoughts?

Reply:
We have an issue #883 that pointed out that we don't have any standalone instruction for IDEs other than VS Code. Can we write an article for that? Then we can add that here as the true "standalone" together with the command line experience on VS Code here.

Reply:
I think the word "standalone" has started to take on a lot of different meanings, in that it can refer to the Q# standalone executable feature, the use of Q# without an explicit host program (whether through IQ# or standalone executables), or the use of the Quantum Development Kit without an editor integration extension. To wit, you can write a Python or .NET interoperability host without an editor extension, which is very much so standalone in the third sense, but I wouldn't think of that as standalone in the first or second sense.

Reply:
@KittyYeungQ @geduardo @cgranade, RE: Chris's comment.
I agree, it's become aligned with so many different situations that it seems to have developed an inability to reasonably describe _any_ particular situation. Perhaps this page might be an opportunity to lay down some concrete wording and remove such ambiguities.

Regardless of what situation ends up being called what, I think a small paragraph describing those nomenclatures wouldn't be jarringly out of place at the beginning of this page, and might be appreciated by the readers.

If we do go that route, from my current understanding I would vote 
- "standalone" be reclaimed by the standalone executable from the .NET CLI, 
- "host program" be reserved for all situations where it's their classical program which calls to Q# (this one's already pretty cut and dried)
- Q# Jupyter Notebooks stay as it's own unique party
- and try in general to avoid confusing phrasing that intermingles these with the other variables involved with the user-Q# interface (e.g. editor integration extensions, etc.).

My understanding could be stunted here, and I trust Chris know best with regard to what pieces of the interface and execution model are independent or inseparable. But in general I'm interested to hear all your thoughts, because I think this is a decent opportunity to remove ambiguity both internally and in the docs